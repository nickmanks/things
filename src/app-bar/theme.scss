
.app-bar-header {
  display: inline-flex;
  justify-content: center;
  padding: 20px 0;
  position: fixed;
  background: #fff;
  z-index: 1040;
  width: 100%;
  max-height: 70px;
  box-shadow: 3px 0 30px rgba(17, 31, 93, .08), 2px 0 5px rgba(27, 27, 43, .09);
}

.app-bar-logo {
  margin-top: -13px;
  display: flex;
}

.app-bar-logo-name {
  margin-top: 12px;
  font-weight: 600;
  margin-left: 15px;
}

.app-bar-logo-img {
  height: 38px;
  margin-left: 77px;
  margin-top: 7px;
}

.app-flex-1 {
  flex-grow:  1;
}

.app-flex-2 {
  flex-grow: 4;
}

.app-flex-3 {
  flex-grow:  1;
  display: flex;
}

.app-bar-search {
  margin-left: -40px;
  margin-top: -10px;
}

.app-bar-archive-icon {
  color: #ABB6BB;
  height: 28px;
  margin-left: 5px;
}

.app-bar-archive-badge {
  height: 24px;
  position: relative;
  right: 13px;
  top: -11px;
}

.flex-dropdown {
  width: 200px;
  display: flex;
  height: 60px;
  margin: auto;
}

.app-bar-sort-dropdown {
  margin-top: -5px;
}

.sort-checkbox {
  margin-left: 20px;
}
