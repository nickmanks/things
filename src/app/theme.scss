
.app-container {
  padding-top: 100px;
  padding-left: 5%;
  padding-right: 5%;
  margin: auto;
}

.app-loading-spinner {
  position: fixed;
  top: 0; right: 0;
  bottom: 0; left: 0;
  background: #fff;
}

.app-spinner {
  left: 50%;
  margin-left: -4em;
  font-size: 10px;
  border: .8em solid rgba(218, 219, 223, 1);
  border-left: .8em solid #007bff;
  animation: spin 1.1s infinite linear;
}

.app-spinner, .app-spinner:after {
  border-radius: 50%;
  width: 8em;
  height: 8em;
  display: block;
  position: absolute;
  top: 50%;
  margin-top: -4.05em;
}

@keyframes spin {
  0% {
    transform: rotate(360deg);
  }

  100% {
    transform: rotate(0deg);
  }
}
