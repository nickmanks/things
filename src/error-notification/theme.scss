
.error-notification-icon-pre {
  margin-right: 25px;
}

.error-notification-icon-post {
  position: absolute;
  right: 34px;
  margin-top: 3px;
  cursor: pointer;
}

.fade-in-error {
  position: fixed;
  top: 0;
  width: 100%;
  margin-top: 70px;
}
