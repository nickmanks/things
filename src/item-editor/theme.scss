
.editor-date-picker {
  margin-bottom: 25px;
  height: auto;
  padding: .5rem 1rem;
  font-size: .95rem;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  border: 1px solid #becad6;
  font-weight: 300;
  will-change: border-color,box-shadow;
  border-radius: .375rem;
  box-shadow: none;
  transition: box-shadow 250ms cubic-bezier(0.27, 0.01, 0.38, 1.06), border 250ms cubic-bezier(0.27, 0.01, 0.38, 1.06);
}

.editor-category {
  margin-bottom: 25px;
}

.editor-created {
  margin-bottom: 25px;
}
