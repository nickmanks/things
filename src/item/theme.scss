
.item-header {
  display: flex;
}

.item-name-input {
  max-width: 97%;
  margin-left: -18px;
  flex-grow: 3;
}

.item-time-pill {
  flex-grow: 1;
  height: 23px;
  margin-top: 8px;
  margin-left: 5px;
  margin-right: -22px;
}

.item-status {
  display: flex;
  text-align: center;
}

.item-status-pill {
  margin: auto;
  cursor: pointer;
}

.item {
  margin-bottom: 25px;
  height: 320px;
  cursor: pointer;
}

// .item:hover {
//   border: 1px solid #3594ff;
// }

.item-body {
  overflow: hidden;
  text-overflow: ellipsis;
}

.outline-item {
  height: 320px;
  box-shadow: none;
  border: 2px dashed #69d7a6;
}

.outline-item-header {
  background-color: white;
}

.outline-item-body {
  margin: auto;
  text-align: center;
}

.outline-item-footer {
  background-color: white;
}

.outline-button {
  margin: auto;
  margin-top: 50%;
}
