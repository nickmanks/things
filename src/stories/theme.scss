
$storybook-container-shadow: 1px 1px 5px 1px rgba(0, 0, 0, .37);

.story-container {
  margin: 3rem;
  box-shadow: $storybook-container-shadow;
  width: 100%;
  height: 65rem;
  border-radius: .4rem;

  > div {
    height: 100%;
    width: 100%;

    > div:first-child {
      height: 100%;
      width: 100%;
    }
  }
}
