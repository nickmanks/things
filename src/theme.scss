
/* root theme imported by bundler */
/* keep colour and theme constant variables here */
