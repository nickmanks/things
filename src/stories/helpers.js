import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'shards-ui/dist/css/shards.min.css';
import './theme.scss';


export function withCustomContainer(backgroundColor) {
  const CustomContainer = (story)=> (
    <div className={'story-container'} style={{backgroundColor}}>
      {story()}
    </div>
  );
  CustomContainer.displayName = 'CustomContainer';
  return CustomContainer;
}

export const storyStore =
  (state)=> createStore(reducers, state, applyMiddleware(thunk));
