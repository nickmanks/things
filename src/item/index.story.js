/* global module */
import React from 'react';
import {storiesOf} from '@storybook/react';
import {storyStore} from '../stories/helpers';
import Item from '.';
import './theme.scss';


storiesOf('Connected/Item', module)
  .add('with in ready item', ()=> {
    const store = storyStore();

    return (
      <Item
        item={{
          name: 'Buy Groceries',
          category: 'Ingredients',
          description: '1x Tomato, 1x Onion, 1x Spaghetti',
          status: 'ready',
          created: '3h'
        }}

        store={store}
      />
    );
  })
  .add('with in progress item', ()=> {
    const store = storyStore();

    return (
      <Item
        item={{
          name: 'Buy Groceries',
          category: 'Ingredients',
          description: '1x Tomato, 1x Onion, 1x Spaghetti',
          status: 'in progress',
          created: '3h'
        }}

        store={store}
      />
    );
  })
  .add('with done item', ()=> {
    const store = storyStore();

    return (
      <Item
        item={{
          name: 'Buy Groceries',
          category: 'Ingredients',
          description: '1x Tomato, 1x Onion, 1x Spaghetti',
          status: 'done',
          created: '3h'
        }}

        store={store}
      />
    );
  });
