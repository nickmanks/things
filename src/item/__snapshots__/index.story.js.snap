// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Connected/Item with done item 1`] = `
<Item item={{"name":"Buy Groceries","category":"Ingredients","description":"1x Tomato, 1x Onion, 1x Spaghetti","status":"done","created":"3h"}} store={{}} onNameChange={"function"} onUpdateStatus={"function"} onEditItem={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Card className={"item"} tag={"div"}>
    <div className={"item card"}>
      <ItemHeader name={"Buy Groceries"} created={"3h"} onNameChange={"function"}>
        <CardHeader tag={"div"}>
          <div className={"card-header"}>
            <div className={"item-header"}>
              <FormInput className={"item-name-input"} value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"}>
                <input value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"} className={"item-name-input form-control"} />
              </FormInput>
              <Badge className={"item-time-pill"} pill={true} theme={"secondary"} outline={false} tag={"span"}>
                <span className={"item-time-pill badge badge-secondary badge-pill"}>
                  NaN days
                </span>
              </Badge>
            </div>
          </div>
        </CardHeader>
      </ItemHeader>
      <ItemBody category={"Ingredients"} description={"1x Tomato, 1x Onion, 1x Spaghetti"} onEditItem={"function"}>
        <CardBody className={"item-body"} onClick={"function"} tag={"div"}>
          <div onClick={"function"} className={"item-body card-body"}>
            <CardSubtitle tag={"h6"}>
              <h6 className={"card-subtitle text-muted"}>
                Ingredients
              </h6>
            </CardSubtitle>
            1x Tomato, 1x Onion, 1x Spaghetti
          </div>
        </CardBody>
      </ItemBody>
      <ItemStatus status={"done"} onUpdateStatus={"function"}>
        <CardFooter tag={"div"}>
          <div className={"card-footer"}>
            <div className={"item-status"}>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  Ready
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  In Progress
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"success"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-success badge-pill"}>
                  Done
                </span>
              </Badge>
            </div>
          </div>
        </CardFooter>
      </ItemStatus>
    </div>
  </Card>
</Item>
`;

exports[`Storyshots Connected/Item with in progress item 1`] = `
<Item item={{"name":"Buy Groceries","category":"Ingredients","description":"1x Tomato, 1x Onion, 1x Spaghetti","status":"in progress","created":"3h"}} store={{}} onNameChange={"function"} onUpdateStatus={"function"} onEditItem={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Card className={"item"} tag={"div"}>
    <div className={"item card"}>
      <ItemHeader name={"Buy Groceries"} created={"3h"} onNameChange={"function"}>
        <CardHeader tag={"div"}>
          <div className={"card-header"}>
            <div className={"item-header"}>
              <FormInput className={"item-name-input"} value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"}>
                <input value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"} className={"item-name-input form-control"} />
              </FormInput>
              <Badge className={"item-time-pill"} pill={true} theme={"secondary"} outline={false} tag={"span"}>
                <span className={"item-time-pill badge badge-secondary badge-pill"}>
                  NaN days
                </span>
              </Badge>
            </div>
          </div>
        </CardHeader>
      </ItemHeader>
      <ItemBody category={"Ingredients"} description={"1x Tomato, 1x Onion, 1x Spaghetti"} onEditItem={"function"}>
        <CardBody className={"item-body"} onClick={"function"} tag={"div"}>
          <div onClick={"function"} className={"item-body card-body"}>
            <CardSubtitle tag={"h6"}>
              <h6 className={"card-subtitle text-muted"}>
                Ingredients
              </h6>
            </CardSubtitle>
            1x Tomato, 1x Onion, 1x Spaghetti
          </div>
        </CardBody>
      </ItemBody>
      <ItemStatus status={"in progress"} onUpdateStatus={"function"}>
        <CardFooter tag={"div"}>
          <div className={"card-footer"}>
            <div className={"item-status"}>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  Ready
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"warning"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-warning badge-pill"}>
                  In Progress
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  Done
                </span>
              </Badge>
            </div>
          </div>
        </CardFooter>
      </ItemStatus>
    </div>
  </Card>
</Item>
`;

exports[`Storyshots Connected/Item with in ready item 1`] = `
<Item item={{"name":"Buy Groceries","category":"Ingredients","description":"1x Tomato, 1x Onion, 1x Spaghetti","status":"ready","created":"3h"}} store={{}} onNameChange={"function"} onUpdateStatus={"function"} onEditItem={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Card className={"item"} tag={"div"}>
    <div className={"item card"}>
      <ItemHeader name={"Buy Groceries"} created={"3h"} onNameChange={"function"}>
        <CardHeader tag={"div"}>
          <div className={"card-header"}>
            <div className={"item-header"}>
              <FormInput className={"item-name-input"} value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"}>
                <input value={"Buy Groceries"} placeholder={"Buy Groceries"} onChange={"function"} className={"item-name-input form-control"} />
              </FormInput>
              <Badge className={"item-time-pill"} pill={true} theme={"secondary"} outline={false} tag={"span"}>
                <span className={"item-time-pill badge badge-secondary badge-pill"}>
                  NaN days
                </span>
              </Badge>
            </div>
          </div>
        </CardHeader>
      </ItemHeader>
      <ItemBody category={"Ingredients"} description={"1x Tomato, 1x Onion, 1x Spaghetti"} onEditItem={"function"}>
        <CardBody className={"item-body"} onClick={"function"} tag={"div"}>
          <div onClick={"function"} className={"item-body card-body"}>
            <CardSubtitle tag={"h6"}>
              <h6 className={"card-subtitle text-muted"}>
                Ingredients
              </h6>
            </CardSubtitle>
            1x Tomato, 1x Onion, 1x Spaghetti
          </div>
        </CardBody>
      </ItemBody>
      <ItemStatus status={"ready"} onUpdateStatus={"function"}>
        <CardFooter tag={"div"}>
          <div className={"card-footer"}>
            <div className={"item-status"}>
              <Badge className={"item-status-pill"} pill={true} theme={"danger"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-danger badge-pill"}>
                  Ready
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  In Progress
                </span>
              </Badge>
              <Badge className={"item-status-pill"} pill={true} theme={"light"} onClick={"function"} outline={false} tag={"span"}>
                <span onClick={"function"} className={"item-status-pill badge badge-light badge-pill"}>
                  Done
                </span>
              </Badge>
            </div>
          </div>
        </CardFooter>
      </ItemStatus>
    </div>
  </Card>
</Item>
`;
